ios/brave-ios/Sources/BraveVPN/Components/Region/BraveVPNRegionListView.swift

```diff
@@ -221,9 +228,12 @@ public struct BraveVPNRegionListView: View {
   }
 
   private func selectDesignatedVPNRegion(at index: Int) {
-    guard !isLoading, let desiredRegion = allRegions[safe: index],
-      desiredRegion.regionName != BraveVPN.selectedRegion?.regionName
-    else {
+    guard !isLoading, let desiredRegion = allRegions[safe: index] else {
+      return
+    }
+
+    // If we're already connected to this region, do nothing.
+    if desiredRegion.regionName == BraveVPN.selectedRegion?.regionName && isVPNEnabled {
       return
```

Comment:
isn't this is the same thing as the old guard?

Reply:
Yes but I added the `isVPNEnabled`. This fixes: https://github.com/brave/brave-browser/issues/45370

Reply:
ah you are right. i missed `isVPNEnabled`